### ISSUE TITLE
Get-CrmConnection fails when 2 or more ADFS servers can be used for authentication

### ISSUE THREAD

**Issue by jakobborg** [seanmcne/Microsoft.Xrm.Data.PowerShell#113]
Hi.

I have encountered a peculiar problem when trying to connect to a CRM 2013 organization where 2 different ADFS servers are available for authentication. It seems, that the supplied HomeRealmUrl is ignored.

It always fails with

```
WARNING: Failed to connect to CRM: BadConnect
WARNING: Failed to connect to CRM: The authentication endpoint Kerberos was not found on the configured Secure Token Service!
Get-CrmConnection : The pipeline has been stopped.
At line:1 char:1
+ Get-CrmConnection -OrganizationName "xxxxxxxx" -HomeRealmUrl "https://adfs.xx ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : SyntaxError: (:) [Get-CrmConnection], PipelineStoppedException
    + FullyQualifiedErrorId : -9,Microsoft.Xrm.Tooling.CrmConnector.Powershell.Commands.GetCrmConnectionCommand

Get-CrmConnection : Failed to connect to CRM: The authentication endpoint Kerberos was not found on the configured Secure Token Service!
At line:1 char:1
+ Get-CrmConnection -OrganizationName "xxxxxxxx" -HomeRealmUrl "https://adfs.xx ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : PermissionDenied: (:) [Get-CrmConnection], Exception
    + FullyQualifiedErrorId : -3,Microsoft.Xrm.Tooling.CrmConnector.Powershell.Commands.GetCrmConnectionCommand
```

**Comment by seanmcne** [comment]
This function is supplied by the core products "xrm tooling", we should be able to get more info from the UI or via the log file for xrm tooling (you can get the log path by supplying the -verbose parameter switch).

**Comment by seanmcne** [comment]
Forgot to include how to show the UI part :) run the same command but include -interactive and that will provide you a user interface for discovery.

**Comment by jakobborg** [comment]
Thanks for getting back to me.

First of all, there's no way to enter the HomeRealmUrl in -Interactive mode, so that is out of the question. I have added a log output from executing the following:

```powershell
$credentials = Get-Credential 
$mspsconnection = Get-CrmConnection `
    -Credential $credentials `
    -OrganizationName "organization" `
    -HomeRealmUrl "https://adfs.xxxxx.com/adfs/services/trust/mex" `
    -ServerUrl "https://organization.xxxxx.com" `
    -Verbose
```

[Microsoft.Xrm.Tooling.CrmConnector.Powershell-2016-05-10.txt](https://github.com/seanmcne/Microsoft.Xrm.Data.PowerShell/files/257289/Microsoft.Xrm.Tooling.CrmConnector.Powershell-2016-05-10.txt)

I also did another experiment ny instantiating the `CrmServiceClient` directly in PowerShell:

```powershell
$password = [Microsoft.Xrm.Tooling.Connector.CrmServiceClient]::MakeSecureString("xxxxxx")
$connection = New-Object Microsoft.Xrm.Tooling.Connector.CrmServiceClient -ArgumentList @(`
    "username", `
    $password, `
    "domain", `
    "https://adfs.xxxxx.com/adfs/services/trust/mex", `
    "organization.xxxxx.com", `
    "443", `
    "organization", `
    $true, `
    $true)
```
Which work perfectly fine.

**Comment by kenakamu** [comment]
So I guess we need to add HomeRealm support in our connect-crmxxxx functions. I will put it in the next version.

**Comment by seanmcne** [comment]
@kenakamu looks like this was fixed in 2.4 - correct?

**Comment by seanmcne** [comment]
Since we've had no traction or comments on this and I believe this was fixed in 2.4 I'm going to close it out.
